constraint/pkg/regorewriter/regorewriter_test.go

```diff
@@ -495,3 +506,256 @@ func TestRegoRewriterErrorCases(t *testing.T) {
 		}
 	}
 }
+
+func TestRegoRewriterRewriteDataRef(t *testing.T) {
+	tests := []struct {
+		name    string
+		prefix  string
+		imports string
+		extern  []string
+		want    string
+	}{
+		{
+			name:    "no rewrite when data ref is an external reference",
+			prefix:  "foo.bar",
+			imports: "data.ext",
+			extern:  []string{"data.ext", "data.inventory"},
+			want:    "data.ext",
+		},
+		{
+			name:    "rewrite for ref is sub of external references",
+			prefix:  "foo.bar",
+			imports: "data.ext",
+			extern:  []string{"data.ext.util"},
+			want:    "data.foo.bar.ext",
+		},
+		{
+			name:    "no rewrite for non data ref",
+			prefix:  "foo.bar",
+			imports: "invalid_data.name",
+			want:    "invalid_data.name",
+		},
+		{
+			name:    "add prefix for data ref",
+			prefix:  "foo.bar",
+			imports: "data.some_ref",
+			want:    "data.foo.bar.some_ref",
+		},
+		{
+			name:    "add prefix same as libs",
+			prefix:  "data.lib",
+			imports: "data.some_ref",
+			want:    "data.data.lib.some_ref",
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			externs := []string{"data.inventory"}
```

Comment:
Repeat this in each `extern` field for each test case. Defaulting makes tests confusing (e.g. what if I want to test behavior when there are no externs?)

Reply:
Removed default extern.